suggest.js

```diff
@@ -109,49 +57,45 @@ function handleSearchSuggest( response ) {
 	if( response == null ) return;
 
 	var ss = $( '#search_suggest' ).empty().show(),
-		str = response.split( "\n" ),
-		searchProviders = getSearchProvidersForLanguage( searchLang )
-	;
+		str = response.split( "\n" );
+
+	// Removing first element because it's the search string itself
+	str.shift();
 
-	ss.append(
-		$( '<div></div>' )
-			.addClass( 'suggest_link' )
-			.text( 'Treffer für ' + searchString )
-	);
+	// Removing the last element because it is always an empty string
+	str.pop();
 
 	$.each( str, function( index, row ) {
-		// skip first row, it's the search string itself
-		if( index === 0 ) return true;
 
 		var entry = row.split( "\t" );
 
-		ss.append(
-			$( '<div></div>' )
-				.addClass( 'suggest_link' )
-				.append(
-					$( '<a></a>' )
-						.attr( 'href', getSearchLink( entry[0], searchLang ) )
-						.addClass( searchString === entry[0] ? 'exact-match' : 'partial-match' )
-						.text( entry[0] )
-				)
-		);
+		if( $.trim( entry ) !== '' ) {
```

Comment:
After `str.pop()`, there shouldn't be any empty string element left.

Reply:
True!